File: src/main/java/net/alexwells/roomery/proxy/ClientProxy.java
package net.alexwells.roomery.proxy;

import net.alexwells.roomery.holder.BlockHolder;
import net.alexwells.roomery.holder.ItemHolder;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.Item;
import net.minecraftforge.client.event.ModelRegistryEvent;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class ClientProxy extends CommonProxy {
    @SuppressWarnings("ConstantConditions")
    @SubscribeEvent
    public void registerModels(ModelRegistryEvent event) {
        ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(BlockHolder.roomHolder), 0, new ModelResourceLocation(BlockHolder.roomHolder.getRegistryName(), "inventory"));

        ModelLoader.setCustomModelResourceLocation(ItemHolder.roomCard, 0, new ModelResourceLocation(ItemHolder.roomCard.getRegistryName(), "inventory"));
    }
}

File: src/main/java/net/alexwells/roomery/proxy/ClientProxy.kt
package net.alexwells.roomery.proxy

import net.alexwells.roomery.mechanic.connector.ConnectorContainerType
import net.alexwells.roomery.mechanic.connector.ConnectorScreen
import net.alexwells.roomery.mechanic.roomholder.RoomHolderContainerType
import net.alexwells.roomery.mechanic.roomholder.RoomHolderScreen
import net.minecraft.client.gui.ScreenManager
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent

class ClientProxy : CommonProxy() {
    @SubscribeEvent
    fun setup(event: FMLCommonSetupEvent) {
        ScreenManager.registerFactory(RoomHolderContainerType) { container, playerInventory, title ->
            RoomHolderScreen(container, playerInventory, title)
        }
        ScreenManager.registerFactory(ConnectorContainerType) { container, playerInventory, title ->
            ConnectorScreen(container, playerInventory, title)
        }
    }
}
